django/db/models/base.py

```diff
@@ -1112,9 +1127,11 @@ def _save_table(
                 base_qs, using, pk_val, values, update_fields, forced_update
             )
             if force_update and not updated:
-                raise DatabaseError("Forced update did not affect any rows.")
+                raise self.NotUpdated("Forced update did not affect any rows.")
```

Comment:
You should raise `cls.NotUpdated` to allow differentiation between different bases.